### PR TITLE
Refactoring and renaming of BoTorchSearcher and BoTorch

### PR DESCRIPTION
**Pull request by mseeger**
*Issue #, if available:*

*Description of changes:*
Simplification and renaming of BoTorchSearcher. This PR will be followed by an extension of the developer tutorial, which uses this searcher as example how to wrap external code. The wrapper should therefore be up-2-date with respect to support from base classes.

Also, one deprecated BoTorch function is replaced.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: syne_tune/optimizer/schedulers/searchers/botorch/botorch_searcher.py [hunk 1]
```diff
@@ -17,34 +17,31 @@
 from syne_tune.try_import import try_import_botorch_message
 
 try:
-    import torch
+    from torch import Tensor, randn_like
     from botorch.models import SingleTaskGP
-    from botorch.fit import fit_gpytorch_model
```
Comment (mseeger) [inline syne_tune/optimizer/schedulers/searchers/botorch/botorch_searcher.py:22]: This function is deprecated

File: syne_tune/optimizer/schedulers/searchers/searcher.py [hunk 2]
```diff
@@ -450,3 +451,84 @@ def _restore_from_state(self, state: dict):
 
     def set_random_state(self, random_state: np.random.RandomState):
         self.random_state = random_state
+
+
+class SearcherWithRandomSeedAndFilterDuplicates(SearcherWithRandomSeed):
```
Comment (mseeger) [inline syne_tune/optimizer/schedulers/searchers/searcher.py:456]: This is code to be introduced in #487, which will be merged before this one here.

File: syne_tune/optimizer/schedulers/searchers/botorch/__init__.py [hunk 3]
```diff
@@ -16,9 +16,11 @@
 
 try:
     from syne_tune.optimizer.schedulers.searchers.botorch.botorch_searcher import (  # noqa: F401
-        BotorchSearcher,
+        BoTorchSearcher,
+        BotorchSearcher,  # deprecated
```
Comment (wesk) [inline syne_tune/optimizer/schedulers/searchers/botorch/__init__.py:20]: Can `BotorchSearcher` be removed entirely in favor of `BoTorchSearcher`?

Comment (wesk) [reply syne_tune/optimizer/schedulers/searchers/botorch/__init__.py:20]: I see -- looks like you are keeping it for backwards compatibility

**Review by wesk — APPROVED** [review]

File: syne_tune/optimizer/schedulers/searchers/botorch/__init__.py (hunk 3 above)
Comment (mseeger) [reply syne_tune/optimizer/schedulers/searchers/botorch/__init__.py:20]: Yes, we can remove it after a while

**Pull request merged**